### PR TITLE
Write DL2 files without images

### LINKED ISSUE #372: DL2 files including images and DL1 parameters and tables
By processing some DL1 files, I realized that the size was too large, when I checked their containment, they had the following datasets:
```
'dl1/event/subarray/mc_shower',
 'dl1/event/subarray/trigger',
 'dl1/event/telescope/image/LST_LSTCam',
 'dl1/event/telescope/parameters/LST_LSTCam',
 'dl2/event/telescope/parameters/LST_LSTCam',
...
```
All the `dl1/*` should have been removed at DL2 level, specially the images that are the ones increasing the size of these files by a large amount. I reproduced this with `v0.4.5` and `master`. I'll look into solving it.

### PR DESCRIPTION
**Pull request by rlopezcoto**
Stripped the images and the copy of the DL1 parameters from the DL2 file. Fixes #372.

### CODE REVIEW

File: lstchain/scripts/lstchain_dl1_to_dl2.py [hunk 1]
```diff
@@ -96,8 +108,19 @@ def main():
...
     outfile = os.path.join(args.outdir, os.path.basename(args.datafile).replace('dl1','dl2'))
 
+    if os.path.exists(outfile):
+        raise IOError(outfile + ' exists, exiting.')    
+
+    dl1_keys = get_dataset_keys(args.datafile)
+    dl1_keys.remove(dl1_params_lstcam_key)
+    dl1_keys.remove(dl1_images_lstcam_key)
+
+    file = open_file(args.datafile)
```
Comment (vuillaut) [inline lstchain/scripts/lstchain_dl1_to_dl2.py:118]: Nice.
Maybe just a context manager here.

Comment (rlopezcoto) [reply lstchain/scripts/lstchain_dl1_to_dl2.py]: Thanks, done.

**Comment by MaxNoe** [comment]
Why is there 0% diff hit? Isn't the dl1_to_dl2 script tested in `lstchain/scripts/tests_test_scripts`?

**Comment by rlopezcoto** [comment]
> Why is there 0% diff hit? Isn't the dl1_to_dl2 script tested in lstchain/scripts/tests_test_scripts?

Yes, it is:
https://github.com/cta-observatory/cta-lstchain/blob/b27f4d91737bc1acff69d12b9c10c3b8c31154b4/lstchain/scripts/tests/test_lstchain_scripts.py#L64

**Comment by MaxNoe** [comment]
Ah, sorry. probably related to #370 

If you merge master, you should see coverage for the script.

**Comment by rlopezcoto** [comment]
There is something wrong with the addition of source dependent parameters that is not related to #371, now for DL1 files I get, apart from the usual keys:
```
'dl1/event/telescope/parameters_src_dependent/LST_LSTCam',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/abounds',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/bounds',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/indices',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/indicesLR',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/mbounds',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/mranges',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/ranges',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/sorted',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/sortedLR',
 'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/zbounds',
.....
``` 

and hdf reading gets blocked when reaches `'dl1/event/telescope/parameters_src_dependent/_i_LST_LSTCam/alpha/indices'`
I moved the test to add the src_dependent parameters to the end, @SeiyaNozaki any idea what is going on?

**Comment by rlopezcoto** [comment]
The problem with the dataframe containing all these subindex seems to be related to:
https://github.com/cta-observatory/cta-lstchain/blob/b27f4d91737bc1acff69d12b9c10c3b8c31154b4/lstchain/io/io.py#L576

because it is also there when we write DL2 files using:
https://github.com/cta-observatory/cta-lstchain/blob/b27f4d91737bc1acff69d12b9c10c3b8c31154b4/lstchain/io/io.py#L594

(that calls `write_dataframe`), maybe the dataframe is not properly written this way?

**Comment by rlopezcoto** [comment]
I tried to get rid of these indexed keys with the help of @MaxNoe, but I was not able to read later the dataframe that I had written through that key. 
I included a fix directly in `lstchain_dl1_to_dl2` not to read/write these keys.

**Comment by vuillaut** [comment]
@rlopezcoto I think you can replace the `write_dl2_dataframe` function by a simple
```
from astropy.io import Table
tab = Table.from_pandas(dataframe)
tab.write(outfile, path=dl2_params_lstcam_key)
```

**Comment by rlopezcoto** [comment]
@vuillaut the problem I was reporting was not caused by the DL2 writing (although the dataframe written also contain these index keys `_i_*`). In any case, the proposed solution I think it does not work to append that dataframe to the current file.
The solution may be to modify `write_dataframe`, @MaxNoe had some ideas about this, maybe he can look into it. In any case I think this PR can be accepted as it is now because it will work even if we keep writing dataframes as we had done until now.

**Comment by rlopezcoto** [comment]
shall we accept this one before reviewing #381 ?

**Comment by MaxNoe** [comment]
I don't mind the order

**Comment by MaxNoe** [comment]
Tests are still failing though due to the changed command line options

**Comment by rlopezcoto** [comment]
> Tests are still failing though due to the changed command line options

Should be fixed now

**Comment by rlopezcoto** [comment]
I removed the check on tables starting with `_i_` since there is a test checking that now:
https://github.com/cta-observatory/cta-lstchain/blob/24d81bcf7615d8aa3c0589d5b0fa6be41634652a/lstchain/io/tests/test_io.py#L27

**Comment by rlopezcoto** [comment]
I messed with the git history... working it out

**Comment by rlopezcoto** [comment]
Re-opened in #384

**Pull request closed without merging**